extensions/src/main/java/org/pentaho/platform/plugin/services/metadata/PentahoMetadataDomainRepository.java

```diff
@@ -807,15 +1065,22 @@ protected Properties loadProperties( final RepositoryFile bundle ) {
    */
   protected RepositoryFile createUniqueFile( final String domainId, final String locale,
                                              final SimpleRepositoryFileData data ) {
-    // Generate a "unique" filename
-    final String filename = UUID.randomUUID().toString();
-
-    // Create the new file
-    final RepositoryFile file =
-      repository.createFile( getMetadataDir().getId(), new RepositoryFile.Builder( filename ).build(), data, null );
+    return createUniqueFile( UUID.randomUUID().toString(), domainId, locale, data );
+  }
 
+  /**
+   * Creates a new repository file (with the supplied data) and applies the proper metadata to this file.
+   *
+   * @param filename filename for new file
+   * @param domainId the Domain id associated with this file
+   * @param locale   the locale associated with this file (or null for a domain file)
+   * @param data     the data to put in the file
+   * @return the repository file created
+   */
+  protected RepositoryFile createUniqueFile( final String filename, final String domainId, final String locale,
```

Comment:
[#3 creating a new Domain object #createUniqueFile and #storeDomain](https://jira.pentaho.com/browse/BACKLOG-36129)